internal/repository/prisma/tenant_limit.go

```diff
@@ -0,0 +1,249 @@
+		Resource: dbsqlc.NullLimitResource{
+			LimitResource: dbsqlc.LimitResourceEVENT,
+			Valid:         true,
+		},
+		LimitValue: sqlchelpers.ToInt(int32(t.config.Limits.DefaultEventLimit)),
+		AlarmValue: sqlchelpers.ToInt(int32(t.config.Limits.DefaultEventAlarmLimit)),
+		Window:     sqlchelpers.TextFromStr(t.config.Limits.DefaultEventWindow.String()),
+	})
+
+	return err
+}
+
+func (t *tenantLimitRepository) createDefaultCronLimit(ctx context.Context, tenantId string) error {
+
+	_, err := t.queries.SelectOrInsertTenantResourceLimit(ctx, t.pool, dbsqlc.SelectOrInsertTenantResourceLimitParams{
+		Tenantid: sqlchelpers.UUIDFromStr(tenantId),
+		Resource: dbsqlc.NullLimitResource{
+			LimitResource: dbsqlc.LimitResourceCRON,
+			Valid:         true,
+		},
+		LimitValue: sqlchelpers.ToInt(int32(t.config.Limits.DefaultCronLimit)),
+		AlarmValue: sqlchelpers.ToInt(int32(t.config.Limits.DefaultCronAlarmLimit)),
+	})
+
+	return err
+}
+
+func (t *tenantLimitRepository) createDefaultScheduleLimit(ctx context.Context, tenantId string) error {
+
+	_, err := t.queries.SelectOrInsertTenantResourceLimit(ctx, t.pool, dbsqlc.SelectOrInsertTenantResourceLimitParams{
+		Tenantid: sqlchelpers.UUIDFromStr(tenantId),
+		Resource: dbsqlc.NullLimitResource{
+			LimitResource: dbsqlc.LimitResourceSCHEDULE,
+			Valid:         true,
+		},
+		LimitValue: sqlchelpers.ToInt(int32(t.config.Limits.DefaultScheduleLimit)),
+		AlarmValue: sqlchelpers.ToInt(int32(t.config.Limits.DefaultScheduleAlarmLimit)),
+	})
+
+	return err
+}
+func (t *tenantLimitRepository) createDefaultWorkerLimit(ctx context.Context, tenantId string) error {
+
+	_, err := t.queries.SelectOrInsertTenantResourceLimit(ctx, t.pool, dbsqlc.SelectOrInsertTenantResourceLimitParams{
+		Tenantid: sqlchelpers.UUIDFromStr(tenantId),
+		Resource: dbsqlc.NullLimitResource{
+			LimitResource: dbsqlc.LimitResourceWORKER,
+			Valid:         true,
+		},
+		LimitValue: sqlchelpers.ToInt(int32(t.config.Limits.DefaultWorkerLimit)),
+		AlarmValue: sqlchelpers.ToInt(int32(t.config.Limits.DefaultWorkerAlarmLimit)),
+		CustomValueMeter: pgtype.Bool{
+			Bool:  true,
+			Valid: true,
+		},
+	})
+
+	return err
+}
+
+func (t *tenantLimitRepository) GetLimits(ctx context.Context, tenantId string) ([]*dbsqlc.TenantResourceLimit, error) {
+	if !t.config.EnforceLimits {
+		return []*dbsqlc.TenantResourceLimit{}, nil
+	}
+
+	limits, err := t.queries.ListTenantResourceLimits(ctx, t.pool, sqlchelpers.UUIDFromStr(tenantId))
+
+	if err != nil {
+		return nil, err
+	}
+
+	// patch custom worker limits
+	for _, limit := range limits {
+
+		if limit.Resource == dbsqlc.LimitResourceWORKER {
+			workerCount, err := t.queries.CountTenantWorkers(ctx, t.pool, sqlchelpers.UUIDFromStr(tenantId))
+			if err != nil {
+				return nil, err
+			}
+			limit.Value = int32(workerCount)
+		}
+
+	}
+
+	return limits, nil
+}
+
+func (t *tenantLimitRepository) CanCreate(ctx context.Context, resource dbsqlc.LimitResource, tenantId string) (bool, error) {
```

Comment:
We probably want to use the cache here, it's not ideal that every single lookup to determine whether to create an event/workflow run involves a database lookup, part of the design here would be to reduce the database load. I'd suggest at least caching for a minute here.